### ISSUE TITLE
[Android] Investigate support for SSLStream below API Level 24

### ISSUE THREAD

**Issue by steveisok** [dotnet/runtime#78715]
The block below will throw a PNSE because [SNIHostName](https://developer.android.com/reference/javax/net/ssl/SNIHostName) and [SSLEngine.setServerNames](https://developer.android.com/reference/javax/net/ssl/SSLParameters#setServerNames(java.util.List%3Cjavax.net.ssl.SNIServerName%3E)) are only supported on API level 24 and above.  

Here is a sample that can throw the PNSE:

```
using var client = new TcpClient("login.sequrix.com", 443);
            using var stream = new SslStream(client.GetStream(), false);
            stream.AuthenticateAsClient(new SslClientAuthenticationOptions()
            {
                TargetHost = "login.sequrix.com"
            });
            stream.Read(new byte[100], 0, 100);
```

We should investigate if there are any alternative approaches.

**Comment by steveisok** [comment]
After looking at this again, it does seem returning `UNSUPPORTED_API_LEVEL` is heavy handed.

https://github.com/dotnet/runtime/blob/563297d76a318f5aef0a4da5a4af2b3eb1676e82/src/native/libs/System.Security.Cryptography.Native.Android/pal_sslstream.c#L484-L488 

It does appear that the java `SSLEngine` can be created with a host name and a port. Not sure how much of the interop we would have to shuffle to accommodate.

**Comment by rgroenewoudt** [comment]
The same problem happens with SocketsHttpHandler:
```
var client = new HttpClient(new SocketsHttpHandler());
await client.GetStringAsync("https://google.com");
````

**Comment by simonrozsival** [comment]
I digged into it and it seems that we can force APIs 21-23 to do what we want, although it's not very clean. I'll try to open a draft PR so that we can have a discussion if those code changes are acceptable or not.

**Comment by marek-safar** [comment]
API23 is 5+ years out of supported version, do we need to support that in net8?

**Comment by simonrozsival** [comment]
There are still quite a few devices running APIs 21-23 in the wild, around 6.1%. It would be easier to drop support for them, but it should be possible to keep supporting them with relatively little effort.

![image](https://user-images.githubusercontent.com/374616/204316432-f29a04d0-849c-420d-a611-38e687855c0a.png)

**Comment by karelz** [comment]
Fixed in main (8.0) in PR #78918 and in 7.0.3 in PR #79280 and in 6.0.14 in PR #79277.
